docs/source/plugin/bot.rst

```diff
@@ -96,6 +96,122 @@ the :meth:`~sopel.bot.SopelWrapper.action` method::
+**bold**, *italic*, _underline_, :strike:`striked`, or ``monospace``. IRC
+formatting works with control code, bytes you can use to tell IRC clients how
+to display some part of the text.
+
+.. seealso::
+
+    If you want to know more about `IRC formatting in general`__ and some of
+    its limitations, the modern IRC documentation may be of interest to you.
+
+    .. __: https://modern.ircdocs.horse/formatting.html
+
+However, dealing with control code yourself is not the most dev-friendly
+approach, hence the :mod:`sopel.formatting` module. It contains various
+functions to help you create stylized texts.
+
+Text styles
+-----------
+
+Let's dive into example, starting with :func:`~sopel.formatting.bold` text::
+
+    from sopel import formatting
+
+    bot.say(formatting.bold('This is some bold text!'))
+
+This will output a line like this:
+
+    [Sopel] **This is some bold text!**
+
+You can use them with Python string formatting::
+
+    emphasis = formatting.bold('important')
+    bot.say('And here is the %s part.' % emphasis)
+
+To get that kind of output:
+
+    [Sopel] And here is the **important** part.
+
+And you can use multiple style functions together, for example with the
+:func:`~sopel.formatting.italic` function::
+
+    word = formatting.italic('very')
+    emphasis = formatting.bold('%s important' % word)
+    bot.say('And here is the %s part.' % emphasis)
+
+To get a result that looks like this:
+
+    [Sopel] And here is the :strongem:`very` **important** part.
+
+Colored styles
+--------------
+
+Colorized text is a bit more complicated, and Sopel tries to provide helpful
+functions and constants for that: the :func:`~sopel.formatting.color` function
+and the :class:`~sopel.formatting.colors` class.
+
+The ``color`` function takes a line of text and a foreground color. It also
+accept an optional background color that uses the same color codes. The color
+codes are listed by the ``colors`` class, and can be used like that::
+
+    bot.say(formatting.color('Red text.', formatting.colors.RED))
+
+This example should produce this output:
+
+    [Sopel] :red:`Red text.`
+
+You can combine colors and styles, like this::
+
+    big = formatting.color(
+        formatting.bold('WARNING'), formatting.colors.RED)
+    small = formatting.italic('warning')
+    bot.say('[%s] This is a %s.' % (big, small))
+
+So you get a similar result as:
+
+    [Sopel] [:strongred:`WARNING`] This is a *warning*.
+
+If you want to prevent spoilers, you could be tempted to take advantage of
+the background color::
+
+    spoiler = formatting.color(
+        'He was the killer.',
+        formatting.colors.BLACK,
+        formatting.colors.BLACK,
+    )
+    bot.say(spoiler)
+
+And expect this (you need to select the text to read it):
+
+    [Sopel] :spoiler:`He was the killer.`
+
+Note that not all combinations of foreground and background colors are happy
+combinations, and you should keep the usage of colors to a minimum.
```

Comment:
```suggestion
Note that not all combinations of foreground and background colors are happy
ones, and you should be mindful of using too many unnecessary colors.
```

One rephrasing option. The phrase that comes to mind is "Use sparingly", but I don't want to be quite _that_ prescriptive.